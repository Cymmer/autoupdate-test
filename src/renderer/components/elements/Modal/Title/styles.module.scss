@import '../../../../scss/variables';
@import '../../../../scss/mixins';
@import '../../../../scss/themes';

@mixin icon($size) {
  padding: 10px;
  border-radius: 100%;
  height: $size;
  width: $size;

  @content;
}

.ModalTitle {
  display: flex;
  align-items: center;

  &___large {
    &_icon {
      @include icon(52px) {
        font-size: 2rem;
        color: $blue-300;
        background-color: transparentize($blue-100, 0.8);
      }
    }

    &_title {
      margin: 0 0 8px 0;
    }

    &_info {
      margin: 0 0 0 16px;

      @include breakpoint(sm) {
        margin: 4px 0 0 16px;
      }
    }
  }

  &___small {
    &_icon {
      @include icon(36px) {
        color: $blue-300;
        background-color: transparentize($blue-100, 0.8);
      }
    }
  }

  &_title {
    margin: 0 0 0 16px;
  }

  &_info {
    display: flex;
    align-items: flex-end;

    @include breakpoint(sm) {
      flex-direction: column;
      align-items: flex-start;
    }
  }
}
