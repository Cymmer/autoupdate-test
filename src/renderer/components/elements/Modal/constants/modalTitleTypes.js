const modalTitleTypes = {
  LARGE: 'large',
  SMALL: 'small',
};

export default modalTitleTypes;
