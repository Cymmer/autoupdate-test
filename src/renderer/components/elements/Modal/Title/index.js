import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import GLOBALS from 'codechum-app-globals';
import styles from './styles.module.scss';

import Icon from '../../Icon';
import Text from '../../Text';
import textTypes from '../../Text/constants/textTypes';

import modalTitleTypes from '../constants/modalTitleTypes';

const ModalTitle = ({
  id,
  iconId,
  type,
  icon,
  className,
  title,
  subject,
  info,
  titleColorClass,
  textClassName,
  iconClassName,
}) => (
  <div className={cn(className, styles.ModalTitle)}>
    <Icon
      className={cn(iconClassName, styles[`ModalTitle___${type}_icon`])}
      icon={icon}
      id={iconId}
    />
    <div className={styles.ModalTitle_info}>
      <div className={styles.ModalTitle_title}>
        {type === modalTitleTypes.SMALL && (
          <Text
            className={textClassName}
            colorClass={titleColorClass || GLOBALS.COLOR_CLASSES.BLUE['300']}
            id={id}
            type={textTypes.HEADING.XXS}
          >
            {title}
          </Text>
        )}
        {type === modalTitleTypes.LARGE && (
          <>
            <Text
              className={styles.ModalTitle___large_title}
              colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['400']}
              id={id}
              type={textTypes.BODY.MD}
            >
              {title}
            </Text>
            <Text
              colorClass={
                titleColorClass || GLOBALS.COLOR_CLASSES.NEUTRAL['700']
              }
              type={textTypes.HEADING.SM}
            >
              {subject}
            </Text>
          </>
        )}
      </div>
      {type === modalTitleTypes.LARGE && (
        <Text
          className={styles.ModalTitle___large_info}
          colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['400']}
          type={textTypes.BODY.SM}
        >
          {info}
        </Text>
      )}
    </div>
  </div>
);

ModalTitle.defaultProps = {
  id: null,
  iconId: null,
  type: modalTitleTypes.LARGE,
  className: null,
  subject: null,
  info: null,
  titleColorClass: null,
  iconClassName: null,
  textClassName: null,
};

ModalTitle.propTypes = {
  iconId: PropTypes.string,
  id: PropTypes.string,
  type: PropTypes.oneOf([modalTitleTypes.LARGE, modalTitleTypes.SMALL]),
  icon: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  className: PropTypes.string,
  subject: PropTypes.string,
  info: PropTypes.string,
  titleColorClass: PropTypes.string,
  iconClassName: PropTypes.string,
  textClassName: PropTypes.string,
};

export default ModalTitle;
