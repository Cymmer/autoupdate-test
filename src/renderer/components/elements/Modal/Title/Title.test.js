import React from 'react';
import { shallow } from 'enzyme';
import Title from './index';
import modalTitleTypes from '../constants/modalTitleTypes';

describe('RENDERING - Modal > Title', () => {
  it('should render with default props', () => {
    const component = shallow(<Title icon="test" title="Test" />);
    expect(component).toMatchSnapshot();
  });

  it('should render the title if type is SMALL', () => {
    const component = shallow(
      <Title icon="test" title="Test" type={modalTitleTypes.SMALL} />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the title, subject, and info if type is LARGE', () => {
    const component = shallow(
      <Title icon="test" title="Test" type={modalTitleTypes.LARGE} />
    );
    expect(component).toMatchSnapshot();
  });
});
