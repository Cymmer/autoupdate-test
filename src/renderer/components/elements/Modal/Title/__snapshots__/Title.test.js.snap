// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - Modal > Title should render the title if type is SMALL 1`] = `
<div
  className="ModalTitle"
>
  <Icon
    className="ModalTitle___small_icon"
    icon="test"
    id={null}
    isPulsating={false}
    style={null}
  />
  <div
    className="ModalTitle_info"
  >
    <div
      className="ModalTitle_title"
    >
      <Text
        className={null}
        colorClass="b___300"
        element={null}
        id={null}
        type="heading___xxs"
      >
        Test
      </Text>
    </div>
  </div>
</div>
`;

exports[`RENDERING - Modal > Title should render the title, subject, and info if type is LARGE 1`] = `
<div
  className="ModalTitle"
>
  <Icon
    className="ModalTitle___large_icon"
    icon="test"
    id={null}
    isPulsating={false}
    style={null}
  />
  <div
    className="ModalTitle_info"
  >
    <div
      className="ModalTitle_title"
    >
      <Text
        className="ModalTitle___large_title"
        colorClass="n___400"
        element={null}
        id={null}
        type="body___md"
      >
        Test
      </Text>
      <Text
        className={null}
        colorClass="n___700"
        element={null}
        id={null}
        type="heading___sm"
      />
    </div>
    <Text
      className="ModalTitle___large_info"
      colorClass="n___400"
      element={null}
      id={null}
      type="body___sm"
    />
  </div>
</div>
`;

exports[`RENDERING - Modal > Title should render with default props 1`] = `
<div
  className="ModalTitle"
>
  <Icon
    className="ModalTitle___large_icon"
    icon="test"
    id={null}
    isPulsating={false}
    style={null}
  />
  <div
    className="ModalTitle_info"
  >
    <div
      className="ModalTitle_title"
    >
      <Text
        className="ModalTitle___large_title"
        colorClass="n___400"
        element={null}
        id={null}
        type="body___md"
      >
        Test
      </Text>
      <Text
        className={null}
        colorClass="n___700"
        element={null}
        id={null}
        type="heading___sm"
      />
    </div>
    <Text
      className="ModalTitle___large_info"
      colorClass="n___400"
      element={null}
      id={null}
      type="body___sm"
    />
  </div>
</div>
`;
